macos/FluentUI/Button/Button.swift

```diff
@@ -185,11 +185,14 @@ open class Button: NSButton {
 
 	open override func mouseDown(with event: NSEvent) {
 		mouseDown = true
+		super.mouseDown(with: event)
+		/// mouseUp doesn't register when super.mouseDown is called
```

Comment:
I don't understand this comment.  Surely we don't a mouseDown event to trigger a mouseUp event?  Or is this saying that somehow calling super.mouseDown causes subsequent mouseUp events to not register?

Reply:
The latter. I'm not sure why though...

Reply:
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/EventOverview/HandlingMouseEvents/HandlingMouseEvents.html#//apple_ref/doc/uid/10000060i-CH6-SW4 explains what's going on under the hood here. Can we just put mouseDown = false after the call to super.mouseDown so we're toggling our FluentUI.Button behavior based on how the super class works instead of sending the mouseUp message to super and potentially modifying the behavior we get from the super class?

Reply:
@anandrajeswaran Thanks for the leads. I couldn't get the UI to work as expected with super.mouseDown. I think NSButton uses the mouse-tracking loop approach to handle mouse events. So, if we want to customize the behavior of any mouse events, the proper way to do it may be to recreate the loop and handle the events ourselves.

Reply:
But we don't want to customize the behavior of mouse events, right? We want existing mouse event behavior with but just to change some things when that status changes. Could we instead just have our mouseDown implementation be:
mouseDown = true
super.mouseDown
mouseDown = false

Reply:
I tried the approach you proposed and other alternatives with `super` but none of them worked for me :/ 

We do want to customize the mouse down and mouse up behaviors for updating the contentTintColor. According to the Apple documentation, the mouse-tracking loop filters and processes all mouse events, which means mouse Up gets processed inside the loop. I agree that we should leverage the super class whenever possible, but unfortunately in this case it cannot suffice the design requirements.

Reply:
What didn't work about this approach? It works locally for me
```
	open override func mouseDown(with event: NSEvent) {
		mouseDown = true
		super.mouseDown(with: event)
		mouseDown = false
	}
```

Reply:
The only issue I had with it was if I hold down the button, the button title disappears but reappears when you drag the cursor outside the button. 
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/67026167/103691930-06d75a80-4f65-11eb-81c7-50f17570fa31.gif)

Reply:
Setting the button typed worked, thanks!